LogsHandler/src/main/java/boot/EntryPoint.java

```diff
@@ -0,0 +1,110 @@
+package boot;
+
+import com.google.gson.JsonObject;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.action.search.SearchRequest;
+import util.InfraUtil;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.HttpURLConnection;
+
+import static java.util.Objects.requireNonNull;
+
+@Path("entry-point")
+public class EntryPoint {
+    public static Logger logger = LogManager.getLogger(EntryPoint.class);
+    private final ElasticSearchHandler elasticSearchClient;
+    private final Producer<String, String> producer;
+    private final WebTarget accountsServiceWebTarget;
+
+    @Inject
+    @Singleton
+    public EntryPoint(ElasticSearchHandler elasticSearchClient, Producer<String, String> producer, WebTarget accountsServiceWebTarget) {
+        this.elasticSearchClient = requireNonNull(elasticSearchClient);
+        this.producer = requireNonNull(producer);
+        this.accountsServiceWebTarget = requireNonNull(accountsServiceWebTarget);
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("create-account")
+    public Response createAccount(String jsonString) {
+        JsonObject jsonObject = InfraUtil.stringToJson(jsonString);
+        Response response = accountsServiceWebTarget.path("create-account")
+                .request(MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.CONTENT_TYPE, "application/json")
+                .post(Entity.json(jsonString));
+        return response;
+    }
+
+    // index message (from post request json) to index
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("index")
+    public Response index(String message, @HeaderParam("X-ACCOUNT-TOKEN") String accountToken) {
+        RecordMetadata metadata = null;
+        JsonObject messageAsJson = InfraUtil.stringToJson(message);
+        messageAsJson.addProperty("User-Agent","Mozilla/5.0 (Macintosh; Intel Mac OS X)");
+        messageAsJson.addProperty("X-ACCOUNT-TOKEN",accountToken);
+
+        Response response = accountsServiceWebTarget.path("account/token")
+                .request(MediaType.APPLICATION_JSON)
+                .header("X-ACCOUNT-TOKEN", accountToken)
+                .get();
+        if (response.getStatus() == 200) {
+            response = sendToKafka(messageAsJson.toString());
```

Comment:
Let's be immutable -
```suggestion
            return sendToKafka(messageAsJson.toString());
```